base_classes/NXobject.nxdl.xml

```diff
@@ -30,5 +30,164 @@
 		This is the base object of NeXus
 	</doc>
         <!--attribute name="name"><doc>name of instance</doc></attribute-->
-</definition>
+	<field name="identifierNAME" type="NX_CHAR" nameType="partial">
+		<doc>
+			An identifier for a (persistent) resource.
 
+			An identifier, provided by some authority, that has been assigned to the
+			real-world object described by this ``NXobject``. To be useful, the identifier
+			must not be reassigned to a different real-world object. It is typical for
+			there to be some mechanism to resolve an identifier, obtaining metadata
+			about the object. Identifiers for which some guarantees exist regarding
+			this resolution process are called persistent identifiers.
+			Persistent identifiers are also known as PIDs.
+		</doc>
+		<attribute name="type" type="NX_CHAR">
+			<doc>The type of identifier used.</doc>
```

Comment:
I think it would make sense to include a comment about using the most specific type when describing the identifier.

For example, all IGSNs are DOIs and all DOIs are Handles; however, an IGSN should have type `IGSN` (and not `DOI` or `Hdl`).

Similarly, an ARK, Purl, ORCID and ROR identifiers should have their corresponding types and should not use the more generic URL identifier.